sale_partner_shipping_invoice_domain/views/sale_order.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2021 Akretion
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record model="ir.ui.view" id="sale_order_form_view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <field name="partner_shipping_id" position="attributes">
+                <attribute
+                        name="domain"
+                >[('id', 'child_of', commercial_partner_id)]</attribute>
+            </field>
+            <field name="partner_invoice_id" position="attributes">
+                <attribute
+                        name="domain"
+                >[('id', 'child_of', commercial_partner_id)]</attribute>
```

Comment:
why not set this domain in the (python) field instead ?

Reply:
+1

Reply:
done